authors/apps/articles/views.py

```diff
@@ -131,3 +135,88 @@ def put(self, request, slug):
                                            context={'request': request},
                                            partial=True)
         return Response(serializer.data, status.HTTP_200_OK)
+
+
+class CommentDetailsView(RetrieveUpdateAPIView):
+    """This class handles the http GET and PUT requests."""
+    permission_classes = (IsAuthorOrReadOnly,)
+    queryset = Comment.objects.filter(is_deleted=False)
+    serializer_class = Comment
+    lookup_field = 'id'
+
+
+class CommentDelete(UpdateAPIView):
+
+    """This class Deletes Comments created by the a particular user"""
+    permission_classes = (IsAuthorOrReadOnly,)
+    queryset = Comment.objects.filter(is_deleted=False)
+    serializer_class = DeleteCommentSerializer
+    lookup_field = 'id'
+
+
+class CommentDetailsView(RetrieveUpdateAPIView):
+    """This class handles the http GET and PUT requests."""
+    permission_classes = (IsAuthorOrReadOnly,)
+    queryset = Comment.objects.filter(is_deleted=False)
+    serializer_class = CommentSerializer
+    lookup_field = 'id'
+
+
+class CreateComment(generics.ListCreateAPIView):
+    """This class Creates comments."""
+    serializer_class = CommentSerializer
+    queryset = Comment.objects.filter(is_deleted=False)
+    permission_classes = (IsAuthenticated, )
+
+    def post(self, request, slug, *args, **kwargs):
+        article = Article.objects.filter(slug=slug, is_deleted=False).first()
```

Comment:
We can use a `.get` as slug is unique

Reply:
If I use this I get an error "Article matching query does not exist"